client_test/helpers_test.go

```diff
@@ -225,21 +223,3 @@ func closeSimulatedChain(t *testing.T, chain chainservice.SimulatedChain) {
 		t.Fatal(err)
 	}
 }
-
-// generateMessageKey generates a ECDSA private key using the given pk bytes, and is idempotent.
-func generateMessageKey(t *testing.T, pk []byte) p2pcrypto.PrivKey {
```

Comment:
This seemed to generate different keys for the same `pk`. I didn't read through `GenerateECDSAKeyPair` but I suspect some time-based information is being used when generating the  key.